### PR TITLE
[issue-46] Replace SSH tunneling classes with library

### LINKED ISSUE #46: Extract SSH logic into a separate library
There is some shared SSH-tunnel code here and in Waggle Dance which needs to be improved. We want to pull the code out to a separate library.

### PR DESCRIPTION
**Pull request by ddcprg**
Fixes #46

### CODE REVIEW

File: circus-train-api/pom.xml [hunk 1]
```diff
@@ -35,6 +35,12 @@
       <artifactId>metrics-core</artifactId>
     </dependency>
 
+    <dependency>
+      <groupId>com.hotels</groupId>
+      <artifactId>hcommon-ssh</artifactId>
+      <scope>provided</scope>
```
Comment (massdosage) [inline circus-train-api/pom.xml]: Why is the scope for this "provided"? This is saying we expect someone else to have put it on the classpath?

Comment (ddcprg) [reply circus-train-api/pom.xml]: this is an error, probably copy&pasta

Comment (ddcprg) [reply circus-train-api/pom.xml]: Actually this doesn't matter because is the API, `circus-train-core` provides the library

Comment (massdosage) [reply circus-train-api/pom.xml]: OK, so can you remove it?

Comment (ddcprg) [reply circus-train-api/pom.xml]: no, I need the marker for the closeable metastore interface

Comment (massdosage) [reply circus-train-api/pom.xml]: So what does "provided" get you here as opposed to not listing a scope at all?

Comment (ddcprg) [reply circus-train-api/pom.xml]: make it not transitive?

Comment (massdosage) [reply circus-train-api/pom.xml]: Yeah, but leaving the scope out would do the same. "provided" you usually use if someone else is going to put the jars on the classpath (as we often do with Hadoop being provided). I just don't understand what "provided" gets us here?

Comment (ddcprg) [reply circus-train-api/pom.xml]: the default is compile which makes the dependency transitive, i.e. any downstream implementation of the API will pull the SSH library which they probably don't need. This is all theoretical

Comment (massdosage) [reply circus-train-api/pom.xml]: Yeah, we don't give any other dependency here this special treatment which is why this one stands out. I can live with it I guess.

**Pull request merged**